lib/literal/array.rb

```diff
@@ -64,6 +64,39 @@ def &(other)
 		end
 	end
 
+	def *(times)
+		case times
+		when String
+			@__value__ * times
+		else
+			__with__(@__value__ * times)
+		end
+	end
+
+	def +(other)
+		case other
+		when ::Array
+			other = Literal::Array(@__type__).new(*other)
```

Comment:
`Literal::Array(@__type__)` this comes up a lot, I wonder if we should assign it to an instance variable in the initializer so we can just reuse the same generic multiple times.

```ruby
@__generic__ = Literal::Array(type)
```

Reply:
Added in 59a6843